From df3861926d0604145810f38a8719984447f33a3c Mon Sep 17 00:00:00 2001
From: Mark Mentovai <mark@mentovai.com>
Date: Tue, 24 Sep 2024 16:11:14 -0400
Subject: [PATCH 3/3] libgcc, Darwin: Don't build legacy libgcc_s.1 on macOS 14
 [PR116809]

d9cafa0c4f0a stopped building libgcc_s.1 on macOS >= 15, in part because
that is required to bootstrap the compiler using the macOS 15 SDK. The
macOS 15 SDK ships in Xcode 16, which also runs on macOS 14. libgcc_s.1
can no longer be built on macOS 14 using Xcode 16 by the same logic that
the previous change disabled it for macOS 15.

	PR target/116809

libgcc/ChangeLog:

	* config.host: Don't build legacy libgcc_s.1 on macOS 14.

Signed-off-by: Mark Mentovai <mark@mentovai.com>
---
 libgcc/config.host | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git libgcc/config.host libgcc/config.host
index 08be426f3fe4..ae4c91dee351 100644
--- libgcc/config.host
+++ libgcc/config.host
@@ -236,7 +236,7 @@ case ${host} in
   esac
   tmake_file="$tmake_file t-slibgcc-darwin"
   case ${host} in
-    x86_64-*-darwin2[0-3]*)
+    x86_64-*-darwin2[0-2]*)
       tmake_file="t-darwin-min-11 t-darwin-libgccs1 $tmake_file"
       ;;
     *-*-darwin2*)
-- 
2.46.2

